pkg/reconciler/common/stages.go

```diff
@@ -34,6 +34,7 @@ type Stages []Stage
 func (stages Stages) Execute(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.KComponent) error {
 	for _, stage := range stages {
 		if err := stage(ctx, manifest, instance); err != nil {
+			instance.GetStatus().MarkInstallFailed(err.Error())
```

Comment:
This could yield confusing behavior, I think. Each "stage" is expected to set the condition that's appropriate for it. So the install stage should be the only stage to ever set the `InstallSucceeded` condition, i.e. call `MarkInstallFailed`. The term "install" in this context means "applied the manifest without error", but we won't be `Ready` until all the other conditions are met, e.g. `DeploymentsAvailable`.

So the way you have this written, the install stage might call `MarkInstallSucceeded` but if the subsequent stage that checks deployments returns an error, this line will overwrite it. That's not giving the user an accurate record of what happened, IMO.

Reply:
Good point!

I've switched these to all simply set `Ready = False` if there's an error, with an "Internal error" message. (And tests added.)

I may see about upstreaming this into `pkg` as well, as it seems like any Reconcile that returns an error should probably signal that in the resource's condition somehow.